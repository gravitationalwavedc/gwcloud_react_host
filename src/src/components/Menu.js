import React from "react";
import {logout} from '../Environment';
import {Link} from 'found';
import Navbar from "react-bootstrap/Navbar";
import Nav from "react-bootstrap/Nav";

const subMenu = (name) => {
  if(name){
    return (
      <Nav>
        <Navbar.Text className="justify-content-end">{name}</Navbar.Text>
        <Nav.Link onClick={() => logout()}>Logout</Nav.Link>
      </Nav>
    );
  };

  return (
    <Nav>
      <Link to="/auth/" exact>Login</Link>
    </Nav>
  );
}

const Menu = ({name}) => {
  const SubMenu = subMenu(name);
  return (
    <Navbar fixed="top">
      <Navbar.Brand className="mr-auto">
        <Link to="/" exact>GW Cloud</Link>
      </Navbar.Brand>
      {SubMenu}
    </Navbar>
  );
}

export default Menu;
