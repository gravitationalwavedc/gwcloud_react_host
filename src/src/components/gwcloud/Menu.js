import React from 'react';
import {logout} from '../../Environment';
import {Link} from 'found';
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';
import { HiOutlineUserCircle, HiOutlineLogout, HiOutlineCode } from 'react-icons/hi';
import GWCloudLogo from '../../assets/gwcloud/images/GWCloud-logo-primary200.png';

const iconStyle = {
    height: '20px',
    margin: '-2px 2px 0 0'
};

const subMenu = (name) => {
    if(name){
        return (
            <Nav>
                <Nav.Link
                    className="justify-content-end mr-3"
                    href="https://gwcloud-python.readthedocs.io/en/latest/gettingstarted.html"
                >
                    <HiOutlineCode style={iconStyle}/> Python API
                </Nav.Link>
                <Navbar.Text className="justify-content-end mr-3">
                    <HiOutlineUserCircle style={iconStyle}/> {name}
                </Navbar.Text>
                <Nav.Link onClick={() => logout()}>
                    <HiOutlineLogout style={iconStyle}/> Logout
                </Nav.Link>
            </Nav>
        );
    }

    return (
        <Nav>
            <Link to="/auth/" exact>Login</Link>
        </Nav>
    );
};

const Menu = ({name}) => {
    const SubMenu = subMenu(name);
    return (
        <Navbar fixed="top">
            <Navbar.Brand className="mr-auto">
                <Link to="/" exact className="navbar-brand-link" data-testid="GWCloudLogo">
                    <img src={GWCloudLogo} style={iconStyle}/>GWCloud
                </Link>
            </Navbar.Brand>
            {SubMenu}
        </Navbar>
    );
};

export default Menu;
