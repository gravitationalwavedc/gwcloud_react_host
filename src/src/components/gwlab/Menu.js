import React from 'react';
import {logout} from '../../Environment';
import {Link} from 'found';
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';
import GWLabLogo from '../../assets/gwlab/images/logo.svg';

const subMenu = (name) => {
    if (name) {
        return (
            <>
                <Nav className="mr-auto">
                    <Nav.Link>Home</Nav.Link>
                    <Nav.Link>Experiments</Nav.Link>
                    <Nav.Link>New Experiment</Nav.Link>
                </Nav>
                <Nav>
                    <Navbar.Text className="justify-content-end mr-3 nav-username">
                        {name}
                    </Navbar.Text>
                    <Nav.Link onClick={() => logout()}>
                    Logout
                    </Nav.Link>
                </Nav>
            </>
        );
    }

    return (
        <Nav>
            <Link to="/auth/" exact>Login</Link>
        </Nav>
    );
};

const Menu = ({name}) => {
    const SubMenu = subMenu(name);
    return (
        <Navbar fixed="top">
            <Navbar.Brand>
                <Link to="/" exact className="navbar-brand-link">
                    <img src={GWLabLogo} />
                </Link>
            </Navbar.Brand>
            {SubMenu}
        </Navbar>
    );
};

export default Menu;
