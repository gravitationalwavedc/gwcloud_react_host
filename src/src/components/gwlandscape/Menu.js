import React from 'react';
import {logout} from '../../Environment';
import {Link} from 'found';
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';
import { HiOutlineUserCircle, HiOutlineLogout } from 'react-icons/hi';
import GWLandscapeLogo from '../../assets/gwlandscape/images/logo.png';

const iconStyle = {
    height: '20px',
    margin: '-2px 2px 0 0'
};

const subMenu = (name) => {
    if(name){
        return (
            <Nav>
                <Navbar.Text className="justify-content-end mr-3">
                    <HiOutlineUserCircle style={iconStyle}/> {name}
                </Navbar.Text>
                <Nav.Link onClick={() => logout()}>
                    <HiOutlineLogout style={iconStyle}/> Logout
                </Nav.Link>
            </Nav>
        );
    }

    return (
        <Nav>
            <Link to="/auth/" exact>Login</Link>
        </Nav>
    );
};

const Menu = ({name}) => {
    const SubMenu = subMenu(name);
    return (
        <Navbar fixed="top">
            <Navbar.Brand className="mr-auto">
                <Link to="/" exact className="navbar-brand-link" data-testid="GWLandscapeLogo">
                    <img src={GWLandscapeLogo} style={iconStyle}/>GWLandscape
                </Link>
            </Navbar.Brand>
            {SubMenu}
        </Navbar>
    );
};

export default Menu;
