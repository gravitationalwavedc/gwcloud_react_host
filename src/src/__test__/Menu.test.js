import React from 'react';
import Menu from '../components/gwcloud/Menu';
import renderer from 'react-test-renderer';


jest.mock('found', () => ({
    Link: component => <a {...component}>{component.children}</a>
}));

describe('the menu component', () => {
    it('displays correctly when logged out', () => {
        expect.hasAssertions();
        const menu = renderer
            .create(<Menu />)
            .toJSON();
        expect(menu).toMatchSnapshot();
    });

    it('displays correctly when logged in', () => {
        expect.hasAssertions();
        const menu = renderer
            .create(<Menu name={'Buffy Summers'} />)
            .toJSON();
        expect(menu).toMatchSnapshot();
    });
});
