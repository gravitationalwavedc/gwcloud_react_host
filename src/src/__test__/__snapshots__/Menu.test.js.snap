// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the menu component displays correctly when logged in 1`] = `
<nav
  className="navbar navbar-expand navbar-light fixed-top"
>
  <span
    className="mr-auto navbar-brand"
  >
    <a
      className="navbar-brand-link"
      exact={true}
      to="/"
    >
      <img
        src="GWCloud-logo-primary200.png"
        style={
          Object {
            "height": "20px",
            "margin": "-2px 2px 0 0",
          }
        }
      />
      GWCloud
    </a>
  </span>
  <div
    className="navbar-nav"
    onKeyDown={[Function]}
  >
    <span
      className="justify-content-end mr-3 navbar-text"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
            "height": "20px",
            "margin": "-2px 2px 0 0",
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.121 17.804A13.937 13.937 0 0112 16c2.5 0 4.847.655 6.879 1.804M15 10a3 3 0 11-6 0 3 3 0 016 0zm6 2a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
       
      Buffy Summers
    </span>
    <a
      className="nav-link"
      data-rb-event-key={null}
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
            "height": "20px",
            "margin": "-2px 2px 0 0",
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17 16l4-4m0 0l-4-4m4 4H7m6 4v1a3 3 0 01-3 3H6a3 3 0 01-3-3V7a3 3 0 013-3h4a3 3 0 013 3v1"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
       Logout
    </a>
  </div>
</nav>
`;

exports[`the menu component displays correctly when logged out 1`] = `
<nav
  className="navbar navbar-expand navbar-light fixed-top"
>
  <span
    className="mr-auto navbar-brand"
  >
    <a
      className="navbar-brand-link"
      exact={true}
      to="/"
    >
      <img
        src="GWCloud-logo-primary200.png"
        style={
          Object {
            "height": "20px",
            "margin": "-2px 2px 0 0",
          }
        }
      />
      GWCloud
    </a>
  </span>
  <div
    className="navbar-nav"
    onKeyDown={[Function]}
  >
    <a
      exact={true}
      to="/auth/"
    >
      Login
    </a>
  </div>
</nav>
`;
